base/glibc-compatibility/musl/getauxval.c

```diff
@@ -17,7 +17,11 @@ static size_t __find_auxv(unsigned long type)
     return (size_t) -1;
 }
 
-__attribute__((constructor)) static void __auxv_init()
+/// __auxv_init should happen BEFORE the first use of getauxval.
+/// but getauxval can be used in other init sections (namely in musl/clock_gettime.c::cgt_init), 
+/// so constructor(101) is needed to prioritize that constructor (0-100 are reserved for gcc-10),
+/// see also: https://stackoverflow.com/questions/11106875/attribute-constructor-call-order-confusion/11198936
+__attribute__((constructor(101))) static void __auxv_init()
```

Comment:
Why do you think it works? I mean I understand that a function with `__attribute__((constructor(101)))` is going to be called before any function with `__attribute__((constructor))` but why is it enough?

Reply:
It doesn't seem like a good solution, `__attribute__((constructor(101)))` is definitely going to be initialized before just `__attribute__((constructor))`, but there are still no guaranties that any of `__attribute__((constructor*))` will be called before the first call of `getauxval()`. So I believe we need lazy initialization here.